### ISSUE TITLE
no-color-literals fails on variable names in maps

### ISSUE THREAD

**Issue by vsiao** [sasstools/sass-lint#651]
```scss
$palette-teal: (
  core: $teal
);
```

**Expected**: no failure
**Actual**: `2:10 warning  Color literals such as 'teal' should not be used as map identifiers`

**Comment by DanPurdy** [comment]
Duplicate of #538 and will be fixed in #647
